samples/transactional-session/aspnetcore-webapi/SqlPersistenceTS_6/WebApplication/Program.cs

```diff
@@ -48,33 +49,56 @@ public static void Main()
             #region txsession-ef-configuration
             .ConfigureServices(c =>
             {
-                // Configure Entity Framework to attach to the synchronized storage session
+                // Configure Entity Framework to attach to the synchronized storage session when required
                 c.AddScoped(b =>
                 {
-                    var session = b.GetRequiredService<ISqlStorageSession>();
-                    var context = new MyDataContext(new DbContextOptionsBuilder<MyDataContext>()
-                        .UseSqlServer(session.Connection)
-                        .Options);
+                    var requiresSessionIntegration = true;
```

Comment:
Why do we need to assume `false`? Is it for the cases where `contextAccessor.HttpContext == null` ?

Reply:
When the http context is null we are in the handler pipeline for the incoming message (control message and others).